gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/SpecCatalog.java

```diff
@@ -110,7 +112,7 @@ public StandardMetrics(final SpecCatalog specCatalog, Optional<Config> sysConfig
       this.totalUpdatedSpecs = new AtomicLong(0);
       this.numActiveSpecs = metricsContext.newContextAwareGauge(NUM_ACTIVE_SPECS_NAME,  ()->{
           long startTime = System.currentTimeMillis();
-          int size = specCatalog.getSpecs().size();
+          int size = specCatalog.getSize();
```

Comment:
incredible!  when we spoke, I had imagined the copious deserialization to originate in the need to  peek inside the `Spec` (e.g. to selectively count by filtering in java-space).

I'm absolutely floored that sampling showed this (teeny-tiny) invocation to occupy 42% of CPU time (on the leader) while accounting for 77.5% of memory allocations

excellent work, arjun!